tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile

```diff
@@ -22,6 +23,7 @@ RUN set -ex && \
     apt-get -y --no-install-recommends install \
     software-properties-common \
     cmake \
+    cpp \
```

Comment:
I find it odd that you specifically need to install the C preprocessor (cpp)? Installing "build-essential" would eliminate a few of these lines.

```
❯ apt-cache depends build-essential                    
build-essential
 |Depends: libc6-dev
  Depends: <libc-dev>
    libc6-dev
  Depends: gcc
  Depends: g++
  Depends: make
    make-guile
  Depends: dpkg-dev
```

Reply:
Yeah great point. I had considered installing `build-essential` to install `cpp`, but I noticed that it installs gcc/g++ along with it (which is gcc-11). The extended `gcc-12` docker image specifically installs gcc-12, so I chose to directly install cpp so we wouldn't muddle two versions of gcc in the same container.
I don't think it's a big blocker though since we do specify `CC=gcc-12` in the environment. Let me know if we're OK with using `build-essential` instead.
* https://github.com/aws/aws-lc/blob/main/tests/ci/docker_images/linux-x86/ubuntu-22.04_gcc-12x/Dockerfile